smriprep/workflows/anatomical.py

```diff
@@ -1049,9 +1086,187 @@ def init_anat_fit_wf(
         ])
         # fmt:on
     elif not t2w:
-        LOGGER.info("No T2w images provided - skipping Stage 7")
+        LOGGER.info("ANAT No T2w images provided - skipping Stage 7")
+    else:
+        LOGGER.info("ANAT Found preprocessed T2w - skipping Stage 7")
+
+    # Stages 8-10: Surface conversion and registration
+    # sphere_reg is needed to generate sphere_reg_fsLR
+    # sphere and sulc are needed to generate sphere_reg_msm
+    # white, pial, midthickness and thickness are needed to resample in the cortical ribbon
+    # TODO: Consider paring down or splitting into a subworkflow that can be called on-demand
+    # A subworkflow would still need to check for precomputed outputs
+    needed_anat_surfs = ["white", "pial", "midthickness"]
+    needed_metrics = ["thickness", "sulc"]
+    needed_spheres = ["sphere_reg", "sphere"]
+
+    # Detect pre-computed surfaces
+    found_surfs = {
+        surf: sorted(precomputed[surf])
+        for surf in needed_anat_surfs + needed_metrics + needed_spheres
+        if len(precomputed.get(surf, [])) == 2
+    }
+    if found_surfs:
+        LOGGER.info(f"ANAT Stage 8: Found pre-converted surfaces for {list(found_surfs)}")
+        surfaces_buffer.inputs.trait_set(**found_surfs)
+
+    # Stage 8: Surface conversion
+    surfs = [surf for surf in needed_anat_surfs if surf not in found_surfs]
+    spheres = [sphere for sphere in needed_spheres if sphere not in found_surfs]
```

Comment:
then we should just be able to do
```suggestion
    surfs = list(needed_anat_surfs - found_surfs.keys())
    spheres = list(needed_spheres - found_surfs.keys())
```